### PR TITLE
[GPU] Fix SDPA shape inference issue

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
Fix SDPA shape inference issue

### Tickets:
 - *CVS-151301*

### CODE REVIEW

**Review by sshlyapn — APPROVED** [review]

**Comment by itikhono** [comment]
@evkotov could double-check this fix using the notebook from the ticket?

**Comment by evkotov** [comment]
It looks like the bug from the ticket fixed, but notebook still doesn't work

File: src/plugins/intel_gpu/src/graph/scaled_dot_product_attention.cpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ std::vector<layout> scaled_dot_product_attention_inst::calc_output_layouts(scale
 
     std::vector<ShapeType> input_shapes;
     for (size_t i = 0; i < impl_param.input_layouts.size(); i++) {
-        input_shapes.push_back(impl_param.get_input_layout(0).get<ShapeType>());
+        input_shapes.push_back(impl_param.get_input_layout(i).get<ShapeType>());
```
Comment (evkotov) [inline src/plugins/intel_gpu/src/graph/scaled_dot_product_attention.cpp:63]: notebook is still doesn't work

**Review by evkotov — CHANGES_REQUESTED** [review]

File: src/plugins/intel_gpu/src/graph/scaled_dot_product_attention.cpp (hunk 1 above)
Comment (itikhono) [reply src/plugins/intel_gpu/src/graph/scaled_dot_product_attention.cpp:63]: this is not a blocker for this fix, please approve

**Review by p-durandin — APPROVED** [review]

**Comment by itikhono** [comment]
@evkotov I've double-checked the notebook step-by-step and it works on GPU after the fix.

**Review by evkotov — APPROVED** [review]

**Comment by itikhono** [comment]
Discussed the issue with @evkotov 
By default, the notebook works on GPU but an additional option (quantization to int8) doesn't. Need to investigate from the transformation side.

**Pull request merged**